src/graphics/aurora/model_kotor.cpp

```diff
@@ -650,6 +744,51 @@ void ModelNode_KotOR::readMesh(Model_KotOR::ParserContext &ctx) {
 	createBound();
 
 	ctx.mdl->seek(endPos);
+
+	_mdxStructSize = mdxStructSize;
+	_vertexCount = vertexCount;
+	_offNodeData = offNodeData;
+}
+
+void ModelNode_KotOR::readSkin(Model_KotOR::ParserContext &ctx) {
+	ctx.mdl->skip(12);
+	uint32 mdxOffsetBoneWeights = ctx.mdl->readUint32LE();
+	uint32 mdxOffsetBoneMappingId = ctx.mdl->readUint32LE();
+	uint32 boneMappingOffset = ctx.mdl->readUint32LE();
+	uint32 boneMappingCount = ctx.mdl->readUint32LE();
+	ctx.mdl->skip(72);
+
+	_mesh->skin = new Skin();
+	_mesh->skin->boneMapping = new float[boneMappingCount];
+	_mesh->skin->boneMappingCount = boneMappingCount;
+	_mesh->skin->boneWeights = new float[_vertexCount * 4];
+	_mesh->skin->boneMappingId = new float[_vertexCount * 4];
+
+	uint32 pos = ctx.mdl->seek(ctx.offModelData + boneMappingOffset);
+	for (uint32 i = 0; i < boneMappingCount; i++)
+		_mesh->skin->boneMapping[i] = ctx.mdl->readIEEEFloatLE();
+	ctx.mdl->seek(pos);
```

Comment:
This should jump to the end of the skin data but not only does it ignore the bone mappings it also ignores all the rest of the skin data whose definition you ignore in line 759.